include/beast/core/flat_streambuf.hpp

```diff
@@ -0,0 +1,297 @@
+        have a capacity of zero, an empty input sequence,
+        and an empty output sequence.
+    */
+    basic_flat_streambuf(basic_flat_streambuf&&);
+
+    /** Move constructor.
+
+        The new object will have the same input sequence
+        and an empty output sequence.
+
+        @note After the move, the moved-from object will
+        have a capacity of zero, an empty input sequence,
+        and an empty output sequence.
+
+        @param alloc The allocator to associate with the
+        stream buffer.
+    */
+    basic_flat_streambuf(basic_flat_streambuf&&,
+        Allocator const& alloc);
+    
+    /** Copy constructor.
+
+        The new object will have a copy of the input sequence
+        and an empty output sequence.
+    */
+    basic_flat_streambuf(basic_flat_streambuf const&);
+
+    /** Copy constructor.
+
+        The new object will have a copy of the input sequence
+        and an empty output sequence.
+
+        @param alloc The allocator to associate with the
+        stream buffer.
+    */
+    basic_flat_streambuf(basic_flat_streambuf const&,
+        Allocator const& alloc);
+    
+    /** Copy constructor.
+
+        The new object will have a copy of the input sequence
+        and an empty output sequence.
+    */
+    template<class OtherAlloc>
+    basic_flat_streambuf(
+        basic_flat_streambuf<OtherAlloc> const&);
+
+    /** Copy constructor.
+
+        The new object will have a copy of the input sequence
+        and an empty output sequence.
+
+        @param alloc The allocator to associate with the
+        stream buffer.
+    */
+    template<class OtherAlloc>
+    basic_flat_streambuf(
+        basic_flat_streambuf<OtherAlloc> const&,
+            Allocator const& alloc);
+
+    /** Construct a flat stream buffer.
+
+        The buffer will have a empty input and output sequences.
+
+        @param size The initial size of the buffer, which must be
+        greater than zero.
+
+        @param limit An optional parameter specifying the maximum
+        of the sum of the input and output sequence sizes that can
+        be allocated. If unspecified, the largest value of `std::size_t`
+        is used.
+    */
+    explicit
+    basic_flat_streambuf(std::size_t size, std::size_t limit = (
+        std::numeric_limits<std::size_t>::max()));
+
+    /** Construct a flat stream buffer.
+
+        The buffer will have a empty input and output sequences.
+
+        @param alloc The allocator to associate with the
+        stream buffer.
+
+        @param size The initial size of the buffer, which must be
+        greater than zero.
+
+        @param limit An optional parameter specifying the maximum
+        of the sum of the input and output sequence sizes that can
+        be allocated. If unspecified, the largest value of `std::size_t`
+        is used.
+    */
+    basic_flat_streambuf(Allocator const& alloc,
+        std::size_t size, std::size_t limit = (
```

Comment:
Good catch! The parens are needed but I wrote them out incorrectly. Should be:
```
(std::numeric_limits<std::size_t>::max)())
```

Rationale: see http://www.boost.org/development/requirements.html
(Scroll down to "Make sure your code compiles in the presence of the min() and max() macros")